arch/Kconfig

```diff
@@ -241,6 +241,20 @@ config XIP
 	  supply a linker command file when building your image. Enabling this
 	  option increases both the code and data footprint of the image.
 
+config XIP_NOT
```

Comment:
that is a weird way to name a kconfig symbol, where is this being used?

Reply:
I know the name is weird. The point of the symbol is to force `CONFIG_EXECUTABLE_RAM=y` if `XIP=n`, since its not possible to do something like:
```
config XIP
    select EXECUTABLE_RAM if !XIP
```
Implementing it as a default on `EXECUTABLE_RAM` would allow the user to disable `EXECUTABLE_RAM` while `XIP` is disabled, which would be an invalid configuration.

Happy to take suggestions on naming or a better way to achieve the goal (choice symbol maybe?)